evm/contracts/HTLC/ChargeableHTLC_ERC.sol

```diff
@@ -52,15 +51,35 @@ contract ChargeableHTLC_ERC is HTLC_ERC {
         IERC20 _token = token;
 
         uint _fee = fee;
-        uint _refillAmount = refillAmount;
+        address _refillAddress = refillAddress;
+
+        IPool pool = IPool(_refillAddress);
+        uint256 _poolCap = pool.poolCap();
+        uint256 _poolBalance = _token.balanceOf(_refillAddress);
+
+        uint256 _withdrawAmount = amount;
+        uint256 _refillAmount;
+
+        if (_poolBalance < _poolCap) {
+            uint256 _poolCapacity = _poolCap - _poolBalance;
+            if(_poolCapacity > 0 && _withdrawAmount > _poolCapacity) {
```

Comment:
Here if `_withdrawAmount <= _poolCapacity` it means that the entire withdraw could go into the pool, so the `refillAmount` could be equal `_withdrawAmount`.
You could add an else statement for this.
Also I think it is unecessary to check `_poolCapacity > 0` since if `_poolBalance < _poolCap` so `_poolCap - _poolBalance > 0`

Reply:
Agreed